src/curve_fitting.py

```diff
@@ -0,0 +1,272 @@
+            if prop not in ["use_custom_style", "custom_style"]:
+                figure_settings.bind_property(prop, canvas, prop, 1 | 2)
+
+        self.curves.bind_property("items", canvas, "items", 2)
+
+        # Scale axis to the data to be fitted, set linear scale
+        for ax in canvas.axes:
+            ax.autoscale()
+            ax.yscale = "linear"
+            ax.xscale = "linear"
+        canvas.highlight_enabled = False
+
+        self.toast_overlay.set_child(canvas)
+        self.present()
+
+    def get_free_variables(self):
+        return (
+            re.findall(
+                r"\b(?!x\b|X\b|sin\b|cos\b|tan\b)[a-wy-zA-WY-Z]+\b",
+                self.equation))
+
+    def on_equation_change(self, _entry, _param):
+        for var in self.get_free_variables():
+            if var not in self.fitting_parameters.get_names():
+                self.fitting_parameters.add_items([FittingParameter(var, 1)])
+        self.fitting_parameters.remove_unused(self.get_free_variables())
+        fit = self.fit_curve()
+        if fit:
+            self.set_entry_rows()
+
+    def on_entry_change(self, entry, _param):
+
+        def is_float(value):
+            try:
+                float(value)
+                return True
+            except ValueError:
+                return False
+
+        for index, row in enumerate(self.fitting_params):
+            param_entries = entry
+
+            # Get the FittingParameterEntry class corresponding to the entry
+            while True:
+                if isinstance(param_entries, FittingParameterEntry):
+                    break
+                param_entries = param_entries.get_parent()
+
+            # Set the parameters for the row corresponding to the entry that
+            # was edited
+            if row == param_entries:
+                initial = param_entries.initial.get_text()
+                lower_bound = param_entries.lower_bound.get_text()
+                upper_bound = param_entries.upper_bound.get_text()
+
+                self.fitting_parameters[index].initial = (
+                    initial if is_float(initial) else 1)
+                self.fitting_parameters[index].lower_bound = (
+                    lower_bound if is_float(lower_bound) else -float("inf"))
+                self.fitting_parameters[index].upper_bound = (
+                    upper_bound if is_float(upper_bound) else float("inf"))
+        self.fit_curve()
+
+    def set_results(self):
+        initial_string = _("Results: \n")
+        buffer_string = initial_string
+        for index, arg in enumerate(self.get_free_variables()):
+            buffer_string += f"\n {arg}: {self.param[index]}"
+
+        self.text_view.get_buffer().set_text(buffer_string)
+        bold_tag = Gtk.TextTag(weight=700)
+        self.text_view.get_buffer().get_tag_table().add(bold_tag)
+
+        start_iter = self.text_view.get_buffer().get_start_iter()
+        end_iter = self.text_view.get_buffer().get_start_iter()
```

Comment:
this also uses get_start_iter()?

Reply:
Yes, it starts both the start and the end at the first character. Then it moves the end position forwards until the first word is covered. This part is put in bold. It's basically a roundabout way to set the "Results" word in bold. Haven't found a cleaner solution, good chance that one exists but didn't have much success there.